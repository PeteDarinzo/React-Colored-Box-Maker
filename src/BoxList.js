import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";


const BoxList = () => {

  const [boxes, setBoxes] = useState([]);

  const addBox = (newBox) => {
    setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }]);
  }

  const deleteBox = (id) => {
    setBoxes(boxes => {
      return boxes.filter(box => box.id !== id);
    });
  }

  return (
    <div>
      <h2>Colored Box Generator</h2>
      <NewBoxForm addBox={addBox} />
      <div>
        {boxes.map(({ height, width, color, id }) =>
          <Box
            height={height}
            width={width}
            color={color}
            key={id}
            id={id}
            deleteBox={deleteBox} />)}
      </div>
    </div>
  );
}

export default BoxList;