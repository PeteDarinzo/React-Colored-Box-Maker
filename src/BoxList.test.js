import React from "react";
import { render, fireEvent, getByLabelText } from '@testing-library/react';
import BoxList from './BoxList';


it("renders a box list without crashing", () => {
  render(<BoxList />);
});

it("matches the box list snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
  const { queryByText, getByLabelText, getByTestId } = render(<BoxList />);

  const heightInput = getByLabelText("Height (pixels)");
  const widthInput = getByLabelText("Width (pixels)");
  const colorInput = getByLabelText("Color");
  const btn = queryByText("Add Box");

  fireEvent.change(heightInput, { target: { value: 100 } });
  fireEvent.change(widthInput, { target: { value: 100 } });
  fireEvent.change(colorInput, { target: { value: "#0000ff" } });
  fireEvent.click(btn);

  const removeBtn = queryByText('X');

  // select the remove button for the blue box
  const blueBox = getByTestId("#0000ff");
  expect(blueBox).toBeInTheDocument();

  // test that box is removed
  fireEvent.click(removeBtn);
  expect(blueBox).not.toBeInTheDocument();

  // test that boxes of differing size and color can be added
  fireEvent.change(heightInput, { target: { value: 200 } });
  fireEvent.change(widthInput, { target: { value: 200 } });
  fireEvent.change(colorInput, { target: { value: "#00ff00" } });
  fireEvent.click(btn);

  const greenBox = getByTestId("#00ff00");
  expect(greenBox).toBeInTheDocument();

  fireEvent.change(heightInput, { target: { value: 50 } });
  fireEvent.change(widthInput, { target: { value: 50 } });
  fireEvent.change(colorInput, { target: { value: "#ff0000" } });
  fireEvent.click(btn);

  const redBox = getByTestId("#ff0000");
  expect(redBox).toBeInTheDocument();

});