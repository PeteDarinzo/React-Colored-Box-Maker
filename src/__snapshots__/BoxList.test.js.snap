// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the box list snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Colored Box Generator
    </h3>
    <form>
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="200"
        min="50"
        name="height"
        type="number"
        value="50"
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="200"
        min="50"
        name="width"
        type="number"
        value="50"
      />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#000000"
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
